### PR TITLE
docs: update `README.md` & `SECURITY.md`

### PR DESCRIPTION
**Pull request by hinto-janai**
Re-writes `README.md` and `SECURITY.md`, other examples:
- https://github.com/monero-project/monero
- https://github.com/paradigmxyz/reth
- https://github.com/ZcashFoundation/zebra

Rendered preview: https://github.com/hinto-janai/cuprate/tree/readme

### CODE REVIEW

File: SECURITY.md
Comment (SyntheticBird45) [inline SECURITY.md:15]: Honestly I think any of us is a trusted maintainer.

File: README.md
Comment (hinto-janai) [inline README.md:2]: Other project's banners are (relatively) 40%~70%

File: README.md
Comment (hinto-janai) [inline README.md:4]: I suggest we decide on a tagline for Cuprate, and that our repo's "About" gets changed to it as well.

Not sure if `alternative` is a good choice for the long-term :)

File: README.md
Comment (hinto-janai) [inline README.md:29]: This `README.md` is pretty sparse for now. We could add sections like `Goals` or `Status` but we'd have to update them frequently.

My vote is for just keeping it minimal until we have something working.

File: SECURITY.md
Comment (hinto-janai) [inline SECURITY.md:15]: I think some criteria for this section is:

- Is an active maintainer
- Wrote/understands most of the codebase
- Can respond quickly
- Can accurately respond to and fix disclosures that require deeper crypto knowledge

So pretty much... only Boog :)

File: README.md
Comment (Boog900) [reply README.md:29]: I agree with keeping it minimal for now

File: README.md
Comment (Boog900) [reply README.md:4]: the immediate alternative is just `Rust` although it is a bit boring

**Comment by Boog900** [comment]
It might be a good idea to add some instructions on how to setup tests, I think the only current requirement is a `monerod` binary at the root of the repo.

File: README.md
Comment (SyntheticBird45) [reply README.md:4]: *Cuprate, an upcoming experimental, modern & secure monero node. Written in Rust * ?

Comment (SyntheticBird45) [reply README.md:4]: It's literally the project description and It's the tagline we used from the beginning

**Comment by hinto-janai** [comment]
> It might be a good idea to add some instructions on how to setup tests

I'm planning another PR expanding `CONTRIBUTING.md`, should I add a testing section there instead?

File: README.md
Comment (hinto-janai) [reply README.md:4]: I meant the tagline used for all time, i.e.:
- `Monero: the secure, private, untraceable cryptocurrency`.
- `Zcash - Financial Privacy in Rust 🦓`

> modern & secure

Not sure how far we should lean into this aspect (at least in the tagline), especially `secure` since that may imply that `monerod` is not secure.

> Written in Rust

I'm 50/50 on this, although, a lot of other projects include this, `arti`, `reth`, `zebra`.

**Comment by Boog900** [comment]
> I'm planning another PR expanding CONTRIBUTING.md, should I add a testing section there instead?

Yeah it probably makes more sense in `CONTRIBUTING`

File: README.md
Comment (SyntheticBird45) [reply README.md:4]: The *secure* originally came from the promised security features, isolation and traffic obfuscation, if this disturb you then what about: *An upcoming experimental, modern and performant Monero node*. no *secure*, we just stick to *modern* since we have a completely different architecture, I don't think its unrealistic to imply that monerod codebase is no more modern, tho it isn't the primary intent of the word.
> I'm 50/50 on this, although, a lot of other projects include this, arti, reth, zebra.

Honestly, I think first thing people wanna know about a rewrite is in which language it has been rewrite to + Rust is a selling point, so definitely have its place in the tagline imo

Comment (hinto-janai) [reply README.md:4]: > An upcoming experimental

Tagline for all time (present tense) so it would maybe be `A modern and performant Monero node`, although one day Cuprate will no longer be 'modern and performant', so I'd vote for a tagline that's more of a description that lasts forever like:

- `Arti - An implementation of Tor, in Rust.`
- `Zcash - Financial Privacy in Rust 🦓`

> Honestly, I think first thing people wanna know about a rewrite is in which language it has been rewrite to

I'm okay with/without so up to @Boog900.

Comment (Boog900) [reply README.md:4]: I don't think we actively _need_ to include Rust in the tagline, if it sounds good I don't really care either way.

---

This is when my suggested name change could've come in handy (REXmr), it would've come with a tagline `Rust Enhanced XMR` AND a T-Rex logo :D

Way too late now though but I mean we could still go with `Rust Enhanced XMR` as the tagline.

**Comment by SyntheticBird45** [comment]
Just saying, I really like horizontal content table instead of vertical one (see [mempool](https://github.com/starkware-libs/mempool) and [old readme](https://github.com/Cuprate/cuprate/blob/db-v0.2/readme.md)). Do you guys have thoughts ?

**Comment by hinto-janai** [comment]
> horizontal content table instead of vertical one

It does look nice but it'll stop working if/when we get more sections. Also it doesn't allow for subsections, e.g.:
```
- [Section](#section)
    - [Sub-section](#sub-section)
```

**Comment by hinto-janai** [comment]
Setting to draft until we decide on a tagline.

**Comment by hinto-janai** [comment]
Re-opening, we can decide on tagline and edit the README later.

**Review by Boog900 — APPROVED** [review]

**Pull request merged**
